From e19200df4f596d1d1221d12faeda85134da561e4 Mon Sep 17 00:00:00 2001
From: Srujana Challa <schalla@marvell.com>
Date: Fri, 19 Feb 2021 19:37:33 +0530
Subject: [PATCH 1281/1921] crypto: cn10k: handle new CPT completion code

Add support to handle new completion code(CPT_COMP_WARN),
which is added with latest microcode version(v1.7).
CPT_COMP_WARN meant only for ipsec opcodes.
CPT_COMP_WARN and CPT_COMP_DONE are same for crypto
opcodes.

Signed-off-by: Srujana Challa <schalla@marvell.com>
Change-Id: Ibe77c789c5a6fc71b58c090cc26a1ec2150fc884
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/46352
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Vidya Sagar Velumuri <vvelumuri@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/crypto/marvell/cn10k/cn10k_cpt_hw_types.h | 2 +-
 drivers/crypto/marvell/cn10k/cn10k_cptvf_reqmgr.c | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/crypto/marvell/cn10k/cn10k_cpt_hw_types.h b/drivers/crypto/marvell/cn10k/cn10k_cpt_hw_types.h
index af553526fe34..627e262b60ad 100644
--- a/drivers/crypto/marvell/cn10k/cn10k_cpt_hw_types.h
+++ b/drivers/crypto/marvell/cn10k/cn10k_cpt_hw_types.h
@@ -139,7 +139,7 @@ enum cn10k_cpt_comp_e {
 	CN10K_CPT_COMP_E_FAULT = 0x02,
 	CN10K_CPT_COMP_E_HWERR = 0x04,
 	CN10K_CPT_COMP_E_INSTERR = 0x05,
-	CN10K_CPT_COMP_E_LAST_ENTRY = 0x06
+	CN10K_CPT_COMP_E_WARN = 0x06
 };
 
 /*
diff --git a/drivers/crypto/marvell/cn10k/cn10k_cptvf_reqmgr.c b/drivers/crypto/marvell/cn10k/cn10k_cptvf_reqmgr.c
index 778bd940d690..b21fb5c97d7e 100644
--- a/drivers/crypto/marvell/cn10k/cn10k_cptvf_reqmgr.c
+++ b/drivers/crypto/marvell/cn10k/cn10k_cptvf_reqmgr.c
@@ -396,6 +396,7 @@ static int cpt_process_ccode(struct pci_dev *pdev,
 		return 1;
 
 	case CN10K_CPT_COMP_E_GOOD:
+	case CN10K_CPT_COMP_E_WARN:
 		/*
 		 * Check microcode completion code, it is only valid
 		 * when completion code is CPT_COMP_E::GOOD
-- 
2.31.1

